pinot-broker/src/main/java/com/linkedin/pinot/broker/api/RequestStatistics.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright (C) 2014-2015 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.broker.api;
+
+import com.linkedin.pinot.common.request.BrokerRequest;
+import com.linkedin.pinot.common.request.QueryType;
+import com.linkedin.pinot.common.response.BrokerResponse;
+
+
+/**
+ * A class to hold the details regarding a request and the statistics.
+ * This object can be used to publish the query processing statistics to a stream for
+ * post-processing at a finer level than metrics.
+ */
+public class RequestStatistics {
```

Comment:
What might be the limits of the data that can be pushed through this?

There are other tidbits of info that I would like to see in here, so would be good to know what the limits (if any) are; for ex, the brokerProcessingTime (current processingtime includes server+broker latency); it would be good to include the server-stats that are normally logged, and perhaps some client side info as well. In anycase, it will be good if this is easily extensible.

Reply:
The limits are not defined here, so we can always add more stuff to this object. Limitations will be (if any) on the streams channel in which events are published. The events will contain data out of this object.  The limitation will be defined by the schema chosen for publishing. The schema may or many not include all members of RequestStatistics.